source/concepts/faq.md

```diff
@@ -59,3 +59,16 @@ There are downsides to relying on [experimental features](https://nixos.org/manu
   Improvements to experimental features have a low priority.
 - The [Nix documentation team](https://nixos.org/community/teams/documentation.html) focuses on improving documentation and learning materials for stable features and common principles.
   When using flakes, you will have to rely more heavily on user-to-user support, third-party documentation, and the source code.
+
+## Are there any known impurities in builds?
+
+Yes, there are:
+
+- CPU architecture—great effort being made to avoid compilation of native instructions in favour of hardcoded supported ones.
+- System's current time/date.
+- Linux kernel parameters e.g. [IPv6 capabilities](https://github.com/NixOS/nix/issues/5615).
+- Timing behaviour of the build system—parallel Make build does not get the correct inputs in some cases.
+- Builds that access the home directory or fetch data from the Internet.
+- When randomness is added.
+
+See [The Nix Hour #12 \[purity at eval-, build- and runtime\]](https://www.youtube.com/watch?v=96jCLfKAa5M&t=913s) for more information and examples.
```

Comment:
Happy to see it being useful! However I don't think it should be added here. Anything worthwhile that I mention during Nix hours should be added to the documentation, such that it's not necessary to link to them. Also, that way I can just link to the docs in future Nix hours :)